### DISCUSSION TITLE
Any possibility to get weblement and its xpath by mouseclick?

### DISCUSSION THREAD

**Discussion by lofti198** [cefsharp/CefSharp#4084] [General]
My goal is to rightclick on any webelement (in the screen winforms version) and get it's xpath or another selector. How can I do that?

**Comment by amaitland** [comment]
If your looking to integrate this directly into your apps, the easiest option is probably to use JavaScript to hook the relevant mouse event and then send a message back to your .Net app.

https://github.com/cefsharp/CefSharp/wiki/Frequently-asked-questions#JSEvent

If you just want to open DevTools from the context menu, like inspect element functionality then that's also an option. From memory DevTools can tell you the xpath.

**Reply by lofti198** [reply]
Thank you so much for the reply!

**Reply by lofti198** [reply]
> If your looking to integrate this directly into your apps, the easiest option is probably to use JavaScript to hook the relevant mouse event and then send a message back to your .Net app.
> 
> https://github.com/cefsharp/CefSharp/wiki/Frequently-asked-questions#JSEvent
> 
> If you just want to open DevTools from the context menu, like inspect element functionality then that's also an option. From memory DevTools can tell you the xpath.

Can I somehow debug or get logs from code, which executing in JS callback?

**Reply by lofti198** [reply]
maybe I can somehow open dev tools in browser?

**Reply by amaitland** [reply]
You can open DevTools programatically see https://github.com/cefsharp/CefSharp/wiki/Trouble-Shooting#javascript-debugging

**Reply by lofti198** [reply]
Hi! Wanted to get MouseEvent object, but failed:
![image](https://user-images.githubusercontent.com/42960488/170666404-ebaac62f-27c2-44e1-97b3-b2118b07141d.png)


in Chrome it works completely OK:
![image](https://user-images.githubusercontent.com/42960488/170666353-c821da4c-465c-4a1f-971e-d4e5c5dc31f1.png)

**Reply by amaitland** [reply]
Typically this is a sign the object is too complex to return. Create your own object that contains just the data you require.
